doc/Programming-guide/Key-platform-features/Endpoint-profiles/index.md

```diff
@@ -6,117 +6,438 @@ nav: /:path/Programming-guide/Key-platform-features/Endpoint-profiles/
-Client-side endpoint profile example
 The following code block provides a simple client-side endpoint profile schema example.
 
+```json
+{  
+   "namespace":"org.myproject",
+   "type":"record",
+   "name":"MyClientSideProfile",
+   "fields":[  
+      {  
+         "name":"country",
+         "type":"string"
+      },
+      {  
+         "name":"city",
+         "type":"string"
+      },
+      {  
+         "name":"age",
+         "type":"int"
+      },
+      {  
+         "name":"hobbies",
+         "type":{  
+            "type":"array",
+            "items":"string"
+         }
+      }
+   ]
+}
+```
+
+The following client-side profile would be compatible with our schema example.
 
 ```json
-{
-    "namespace": "org.myproject",
-    "type": "record",
-    "name": "MyClientSideProfile",
-    "fields": [
-        { "name": "country", "type": "string" },
-        { "name": "city", "type": "string" },
-        { "name": "age", "type": "int" },
+{  
+   "country":"US",
+   "city":"San Francisco",
+   "age":32,
+   "hobbies":[  
+      "skydiving",
+      "hiking"
+   ]
+}
+```
+The schema structure from our example allows filtering the endpoints by the owner's country (for example, to show only US news), city (for example, to push weather 
+notifications only for the specified cities), age (for example, to apply age restrictions), and hobbies (for example, to push football scores for those whose hobby list 
+contains "football"). It is allowed to create complex filtering conditions by combining as many filtering conditions as needed.
+
+### Client-side endpoint profile update ###
+
+Endpoint profile information changes as the result of the client operation or user's actions, it is the client implementation responsibility to update the 
+profile via the endpoint SDK API calls. The endpoint SDK detects profile changes by comparing the new profile hash against the previously persisted one. Should there be a change, 
+the endpoint profile management module passes it to the Operations server, which in turn updates the endpoint profile information in the database and revises the endpoint 
+groups membership.
+
+Think about the client-side profile schema as of a structured data set of your endpoint application that will later be available to you in Kaa server and may change 
+due to your client application logic or device state.
+You can configure your own client-side profile schema using the 
+[Admin UI](#setting-client-side-endpoint-profile-update-from-admin-ui) or [REST API]({{root_url}}Programming-guide/Server-REST-APIs). 
+For the purpose of this guide we will use a fairly abstract client-side profile schema shown below.
```

Comment:
It is very strange that we use two sample profile schemes in one page. Please use this schema in "Client-side endpoint profile example" section